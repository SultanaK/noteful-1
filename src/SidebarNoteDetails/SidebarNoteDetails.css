#back__link{
    background-color: #ffcdbf;
} 

.sidebar p{
    font-family: 'Fresca', sans-serif;
    background-color: lightcoral;
    margin: 0;
}

p.sidebar__error{
    background-color: #ffcdbf;
    padding: 10px;
}