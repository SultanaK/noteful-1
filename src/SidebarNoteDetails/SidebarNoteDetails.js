import React from 'react';
import { NavLink } from 'react-router-dom';
import './SidebarNoteDetails.css';

export default class SidebarNoteDetails extends React.Component{
    render(){
        const folderId = this.props.notes.find(note => note.id === this.props.noteId).folderId;
        const folderName = this.props.folders.find(folder => folder.id === folderId).name;

        return(
            <div className="sidebar">
                <NavLink to="/">Go Back</NavLink>
                <p>{folderName}</p>
            </div>
        )
    }
}