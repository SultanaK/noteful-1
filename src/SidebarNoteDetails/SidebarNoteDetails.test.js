import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { BrowserRouter } from 'react-router-dom';
import SidebarNoteDetails from './SidebarNoteDetails';

describe('Sidebar when note is clicked', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter>
          <SidebarNoteDetails folders={sampleFolders} folderId={sampleFolders[0].id}/>
        </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI correctly', () => {
        const tree = renderer
          .create(<BrowserRouter><SidebarNoteDetails folders={sampleFolders} folderId={sampleFolders[0].id}/></BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    })
})

const sampleFolders = [
    {
      "id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Important"
    },
    {
      "id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Super"
    },
    {
      "id": "b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Spangley"
    }
  ];