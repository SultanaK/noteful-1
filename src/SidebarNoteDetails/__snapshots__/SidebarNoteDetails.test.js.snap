// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar when note is clicked renders UI correctly 1`] = `
<div
  className="sidebar"
>
  <p>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-folder-open fa-w-18 "
      data-icon="folder-open"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M572.694 292.093L500.27 416.248A63.997 63.997 0 0 1 444.989 448H45.025c-18.523 0-30.064-20.093-20.731-36.093l72.424-124.155A64 64 0 0 1 152 256h399.964c18.523 0 30.064 20.093 20.73 36.093zM152 224h328v-48c0-26.51-21.49-48-48-48H272l-64-64H48C21.49 64 0 85.49 0 112v278.046l69.077-118.418C86.214 242.25 117.989 224 152 224z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Important
  </p>
  <a
    aria-current="page"
    aria-label="back button"
    className="active"
    href="/"
    id="back__link"
    onClick={[Function]}
    style={Object {}}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-arrow-alt-circle-left fa-w-16 "
      data-icon="arrow-alt-circle-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zm116-292H256v-70.9c0-10.7-13-16.1-20.5-8.5L121.2 247.5c-4.7 4.7-4.7 12.2 0 16.9l114.3 114.9c7.6 7.6 20.5 2.2 20.5-8.5V300h116c6.6 0 12-5.4 12-12v-64c0-6.6-5.4-12-12-12z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Go Back
  </a>
</div>
`;
